Replace `recycleFile()` in `spo file copy` and `spo file move`, closes: #3370

MathijsVerbeeck opened the pull request:
Replaces the occurences of the `recycleFile()` command by `Cli.executeCommandWithOutput()`

Closes: #3370

milanholemans commented:
Thank you @MathijsVerbeeck , we'll review it ASAP!

Jwaegebaert commented on src/m365/spo/commands/file/file-copy.ts:
```diff
@@ -169,58 +173,40 @@ class SpoFileCopyCommand extends SpoCommand {
+      await Cli.executeCommand(removeCommand as Command, { options: { ...removeOptions, _: [] } });
+    }
+    catch (err: any) {
+      if (err.error !== undefined && err.error.message !== undefined && err.error.message.includes('does not exist')) {
+        
+      } 
+      else {
+        throw err;
+      }
```
Let's try to remove the else here and work with a single if
```suggestion
      if (!(err.error !== undefined && err.error.message !== undefined && err.error.message.includes('does not exist'))) {
        throw err;        
      }

Jwaegebaert commented on src/m365/spo/commands/file/file-move.ts:
```diff
@@ -170,57 +174,40 @@ class SpoFileMoveCommand extends SpoCommand {
+      await Cli.executeCommand(removeCommand as Command, { options: { ...removeOptions, _: [] } });
+    } 
+    catch (err: any) {
+      if (err.error !== undefined && err.error.message !== undefined && err.error.message.includes('does not exist')) {
+        
+      }
+      else {
+        throw err;
+      }
```
Let's try to remove the else here and work with a single if

Jwaegebaert commented on src/m365/spo/commands/file/file-rename.ts:
```diff
@@ -137,26 +130,28 @@ class SpoFileRenameCommand extends SpoCommand {
+      await Cli.executeCommand(removeCommand as Command, { options: { ...removeOptions, _: [] } });
+    } 
+    catch (err: any) {
+      if (err.error !== undefined && err.error.message !== undefined && err.error.message.includes('does not exist')) {
+        
+      }
+      else {
+        throw err;
+      }
```
Let's try to remove the else here and work with a single if

Jwaegebaert reviewed (changes requested):
Great job @MathijsVerbeeck! We're almost there. I left a few suggestions you can take a look at. 

I also have an extra question. In the original PR we returned the output of the method `recycleFile`. Is there a reason why this has been changed?

MathijsVerbeeck commented:
Hi @Jwaegebaert, Thanks for reviewing my PR.

I've rewritten the changes. I have also noticed that, by changing to executeCommand instead of executeCommandWithOutput that. The reason that I've changed this to executeCommand is because that we didn't really use the output of the "executeCommandWithOutput", and the executeCommand will just throw an error and thus go into the catch whenever it fails.

Jwaegebaert reviewed (approved):
Great job @MathijsVerbeeck!!!

Thanks so much for your hard work. 🚀

Jwaegebaert commented:
Merged manually
